### PR TITLE
Add some TS types to feeds

### PR DESCRIPTION
**Pull request by HenryNguyen5**

### CODE REVIEW

File: feeds/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import feeds from './feeds.json'
+
+export interface FeedConfig {
+  contractAddress: string
+  contractType: 'aggregator'
+  valuePrefix: '$' | 'Ξ'
```
Comment (rupurt) [inline feeds/src/config.ts:6]:
```suggestion
  valuePrefix: '$' | 'Ξ' | '£' | '¥'
```

File: feeds/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  heartbeat?: number
+  compareOffchain?: string
+  health_price?: string
+  multiply?: string
+  sponsored?: string[]
+  decimalPlaces?: number
+  contractVersion?: 1 | 2 | 3
+}
+
+export function getFeedsConfig(): FeedConfig[] {
```
Comment (rupurt) [inline feeds/src/config.ts:24]: 👍 this is good prep for some upcoming stories

File: feeds/src/components/listing/GridItem.test.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ const AllTheProviders: React.FC = ({ children }) => {
 const feed = partialAsFull<FeedConfig>({
   name: 'pair name',
   path: '/link',
-  valuePrefix: 'prefix',
+  valuePrefix: '$',
```
Comment (hellobart) [inline feeds/src/components/listing/GridItem.test.tsx:25]: Please change `expect(container).toHaveTextContent('prefix')` to ` expect(container).toHaveTextContent('$')`

File: feeds/src/components/listing/GridItem.tsx [hunk 4]
```diff
@@ -80,44 +81,48 @@ interface CompareOffchainProps {
...
-    return sliced.map((name: any, i: number) => [
-      i > 0 && ', ',
-      <span key={name}>{name}</span>,
-    ])
+    return (
+      <>
+        {sliced.map((name, i) => (
+          <Fragment key={name}>
+            {i > 0 ? ', ' : ''}
+            <span key={name}>{name}</span>
```
Comment (hellobart) [inline feeds/src/components/listing/GridItem.tsx:109]: Is `key={name}` needed here?

File: feeds/src/components/listing/GridItem.tsx [hunk 5]
```diff
@@ -54,13 +55,13 @@ export const GridItem: React.FC<Props> = ({
             </>
           )}
         </div>
-        {feed.sponsored && feed.sponsored.length > 0 && (
+        {sponsors.length > 0 && (
```
Comment (hellobart) [inline feeds/src/components/listing/GridItem.tsx:58]: This line breaks the tests as we show sponsors only when answers are fetched (`listingAnswer`)

**Comment by HenryNguyen5** [comment]
> Hey Henry, Could you take a look at the `tests`? A few tests are failing because the relative imports are missing in some places. Do you have any idea why this happens?

Ill take a look, thanks for bringing it up

**Comment by rupurt** [comment]
@HenryNguyen5 looks like there's some extra config required to import `config`.

File: feeds/src/components/listing/GridItem.tsx (hunk 5 above)
Comment (HenryNguyen5) [reply feeds/src/components/listing/GridItem.tsx:58]: Ah nice catch, was typo on my side

**Pull request merged**